data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,577 @@
+
+In OCaml, functions are values. In comparison to other mainstream languages, this creates a richer picture between expressions, values, and names. 
+
+> Note: This tutorial uses the OCaml toplevel UTop. The ["Introduction to the OCaml Toplevel" guide](https://ocaml.org/docs/toplevel-introduction) covers how to use UTop. It would benefit the reader to write variations around the examples provided to strengthen understanding. The topics discussed are not limited to interactive execution of OCaml expressions. However, we believe they are easier to understand within the dynamics of interaction with the OCaml interpreter.
+
+**Prerequisites**: This is an intermediate-level tutorial. Ensure you have have [completed the “Get Started” series](https://ocaml.org/docs/installing-ocaml) before proceeding with this tutorial.
+
+<!--
+When presenting OCaml or another functional programming language, it is often said: “Functions are treated as first-class citizens.” Without further explanation or context, this may not be helpful (it wasn't to me). The goal of this tutorial is to acquire the capabilities implied and entailed by that sentence. In turn, this should explain it:
+- Write all kinds of definitions, for all kinds of values, in all different ways
+- Accurately assess the scope of any definition
+- Avoid being fooled by shadowed definitions and scopes
+- Explain what a closure is and why it is so named
+- Partially apply functions, create closures, and pass them as values
+- Curry and uncurry functions
+- Use the `unit` type as a function parameter or return value, when needed
+-->
+
+## Non-Function Values
+
+### Values
+
+WIP: Rewrite
+
+Like most functional programming languages, OCaml is an [expression-oriented programming language](https://en.wikipedia.org/wiki/Expression-oriented_programming_language). It essentially has a single statement: the global `let`. Programmed computations take place throughout the evaluation of expressions. On termination, a value is produced. Here are a few examples of expressions, their type, and the resulting values:
+
+```ocaml
+# "Everything has a value, every value has a type";;
+- : string = "Everything is a value, every value has a type"
+
+# 2 * 21;;
+- : int = 42
+
+# int_of_float;;
+- : float -> int = <fun>
+
+# int_of_float (3.14159 *. 2.0);;
+- : int = 6
+
+# fun x -> x * x;;
+- : int -> int = <fun>
+
+# ();;
+- : unit = ()
+
+# print_endline;;
+- : string -> unit = <fun>
+
+# print_endline "Hello!";;
+Hello!
+- : unit
+```
+
+An expression type (before evaluation) and the value type it produces (after computation) are the same. This allows the compiler to avoid runtime type checks in binaries. In OCaml, the compiler removes type information, so it's not available at runtime. In programming theory, this is called [subject reduction](https://en.wikipedia.org/wiki/Subject_reduction).
```

Comment:
```suggestion
An expression's type (before evaluation) and its resulting value's type (after computation) are the same. This allows the compiler to avoid runtime type checks in binaries. In OCaml, the compiler removes type information, so it's not available at runtime. In programming theory, this is called [subject reduction](https://en.wikipedia.org/wiki/Subject_reduction).
```

An expression has a type. The type belongs to the expression, so I guess it should be possessive.